var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -37,92 +38,120 @@ class Neuron(Package):
 
     depends_on('mpi',         when='+mpi')
     depends_on('ncurses',     when='~cross-compile')
     depends_on('python@2.6:', when='+python', type=('build', 'link', 'run'))
+    depends_on('py-pytest',   when='+python+tests')
     # Numpy is required for Vector.as_numpy()
     # depends_on('py-numpy',    when='+python', type='run')
+    depends_on('py-cython',   when='+python', type=('build', 'link', 'run'))
     depends_on('tau',         when='+profile')
 
+    conflicts('+cmake',   when='@0:7.8.0a')
     conflicts('~shared',  when='+python')
     conflicts('+pysetup', when='~python')
     conflicts('+rx3d',    when='~pysetup')
 
-    _default_options = ['--without-iv',
-                        '--without-x']
-    _specs_to_options = {
-        '+cross-compile': ['cross_compiling=yes',
-                           '--without-readline',
-                           '--without-memacs',
-                           '--without-nmodl'],
-        '~python':    ['--without-nrnpython'],
-        '~pysetup':   ['--disable-pysetup'],
-        '+mpi+multisend': ['--with-multisend'],
-        '~rx3d':      ['--disable-rx3d'],
-        '~mpi':       ['--without-paranrn'],
-        '+mpi':       ['--with-paranrn=dynamic'],
-        '~shared':    ['--disable-shared'],
-        '+binary':    ['linux_nrnmech=no'],
-    }
-
-    filter_compiler_wrappers('*/bin/nrniv_makefile')
-    filter_compiler_wrappers('*/bin/nrnmech_makefile')
-    filter_compiler_wrappers('*/bin/nrnoc_makefile')
-    filter_compiler_wrappers('*/share/nrn/libtool')
-
-    def get_neuron_archdir(self):
-        """Determine the architecture-specific neuron base directory.
-
-        Instead of recreating the logic of the neuron's configure
-        we dynamically find the architecture-specific directory by
-        looking for a specific binary.
-        """
-        file_list = find(self.prefix, '*/bin/nrniv_makefile')
-        # check needed as when initially evaluated the prefix is empty
-        if file_list:
-            neuron_archdir = os.path.dirname(os.path.dirname(file_list[0]))
-        else:
-            neuron_archdir = self.prefix
-
-        return neuron_archdir
-
-    def patch(self):
-        # aclocal need complete include path (especially on os x)
-        pkgconf_inc = '-I %s/share/aclocal/' % (self.spec['pkgconfig'].prefix)
-        libtool_inc = '-I %s/share/aclocal/' % (self.spec['libtool'].prefix)
-        newpath = 'aclocal -I m4 %s %s' % (pkgconf_inc, libtool_inc)
-        filter_file(r'aclocal -I m4', r'%s' % newpath, "build.sh")
+    # ==============================================
+    # ==== CMake build system related functions ====
+    # ==============================================
+    @when('+cmake')
+    def cmake_args(self):
+        def cmake_enable_option(name, spec_requiremement):
+            value = 'TRUE' if spec_requiremement in self.spec else 'FALSE'
+            return '-DNRN_ENABLE_' + name + ':BOOL=' + value
+        args = [
```

Comment:
Nice! Could maybe be improved with
`args = [cmake_enable_option(setting, variant) for (setting, variant) in [('SHARED', '+shared'), (…)…]]`. Maybe…

Reply:
I refactored this a bit to avoid writing `SHARED` for `+shared`, etc. Check it out if you want